src/main/java/com/owncloud/android/lib/resources/files/ChunkedFileUploadRemoteOperation.java

```diff
@@ -315,4 +316,12 @@ private PutMethod createPutMethod(String uriPrefix) {
 
         return putMethod;
     }
+
+    private int CalcAssembleTimeout(File file) {
+        final double threeMinutes = 3.0 * 60 * 1000;
```

Comment:
I don't think this needs to be a `double`, an `int` should work fine.

Reply:
In the original version, this was a double to prevent `threeMinutes * file.length() / 1e9` from truncating to full Gb, but by declaring `fileSizeInGb` as double, this has indeed become unnecessary.